src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -69,6 +84,186 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+            // Source: https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-findlocalename
+            hr = localizedFamilyNames->FindLocaleName(L"en-us", &localizedNameIndex, &exists);
+        }
+        if (!exists)
+        {
+            // failed to find the correct locale, using the first one
+            localizedNameIndex = 0;
+        }
+
+        // get the localized name
+        UINT32 nameLength;
+        THROW_IF_FAILED(localizedFamilyNames->GetStringLength(localizedNameIndex, &nameLength));
+
+        localizedName.resize(nameLength);
+        THROW_IF_FAILED(localizedFamilyNames->GetString(localizedNameIndex, localizedName.data(), nameLength + 1));
+
+        // now get the nameID
+        hr = localizedFamilyNames->FindLocaleName(L"en-us", &nameIDIndex, &exists);
+        if (FAILED(hr) || !exists)
+        {
+            // failed to find it, using the first one
+            nameIDIndex = 0;
+        }
+
+        // get the nameID
+        THROW_IF_FAILED(localizedFamilyNames->GetStringLength(nameIDIndex, &nameLength));
+        nameID.resize(nameLength);
+        THROW_IF_FAILED(localizedFamilyNames->GetString(nameIDIndex, nameID.data(), nameLength + 1));
+
+        if (!nameID.empty() && !localizedName.empty())
+        {
+            return make<Font>(nameID, localizedName);
+        }
+        return nullptr;
+    }
+
+    IObservableVector<Editor::Font> ProfileViewModel::CompleteFontList() const noexcept
+    {
+        return _FontList;
+    }
+
+    IObservableVector<Editor::Font> ProfileViewModel::MonospaceFontList() const noexcept
+    {
+        return _MonospaceFontList;
+    }
+
+    // Method Description:
+    // - Searches through our list of monospace fonts to determine if the settings model's current font face is a monospace font
+    // - NOTE: This is information stored from DWrite in _UpdateFontList()
+    bool ProfileViewModel::UsingMonospaceFont() const noexcept
+    {
+        bool result{ false };
+        const auto currentFont{ FontFace() };
+        for (const auto& font : _MonospaceFontList)
+        {
+            if (font.LocalizedName() == currentFont)
+            {
+                result = true;
+            }
+        }
+        return result;
+    }
+
+    // Method Description:
+    // - Determines whether we should show the list of all the fonts, or we should just show monospace fonts
+    bool ProfileViewModel::ShowAllFonts() const noexcept
+    {
+        // - _ShowAllFonts is directly bound to the checkbox. So this is the user set value.
+        // - If we are not using a monospace font, show all of the fonts so that the ComboBox is still properly bound
+        return _ShowAllFonts || !UsingMonospaceFont();
+    }
+
+    void ProfileViewModel::ShowAllFonts(const bool& value)
+    {
+        if (_ShowAllFonts != value)
+        {
+            _ShowAllFonts = value;
+            _NotifyChanges(L"ShowAllFonts");
+        }
     }
```

Comment:
an aside: almost starting to feel like PVM warrants its own file

Reply:
_agreed_. I think this'll be a part of #9207.